yt_dlp/extractor/kanopy.py

```diff
@@ -0,0 +1,127 @@
+                data=json.dumps(
+                    {
+                        'credentialType': 'email',
+                        'emailUser': {'email': username, 'password': password},
+                    }
+                ).encode(),
+            )
+            self._ACCESS_TOKEN = try_call(lambda: access_json['jwt'])
+            if self._ACCESS_TOKEN is None:
+                self.report_warning('Failed to get Access token')
+            else:
+                self.headers.update({'Authorization': 'Bearer %s' % self._ACCESS_TOKEN})
+                self._USER_ID = try_call(lambda: access_json['userId'])
+        except ExtractorError as e:
+            self.report_warning(e)
+
+    def _real_extract(self, url):
+        lang, institution, video_id = self._match_valid_url(url).groups()
+
+        memberships = traverse_obj(
+            self._download_json(
+                self._API_BASE_URL + f'memberships?userId={self._USER_ID}',
+                None,
+                headers=self.headers,
+            ),
+            ('list'),
+            list,
+        )
+
+        domain_id = try_call(
+            lambda: self._match_membership(memberships, institution)['domainId']
+        )
+        if not domain_id:
+            raise ExtractorError(f'Could not match {institution} to a membership')
+
+        video_info = self._download_json(
+            self._API_BASE_URL + f'videos/{video_id}?domainId={domain_id}',
+            video_id,
+            headers=self.headers,
+        )
+
+        streams = self._download_json(
+            self._API_BASE_URL + 'plays',
+            video_id,
+            headers=self.headers,
+            data=json.dumps(
+                {
+                    'domainId': domain_id,
+                    'userId': self._USER_ID,
+                    'videoId': video_id,
+                }
+            ).encode(),
+        )
+
+        formats, subtitles = [], {}
+        for manifest in traverse_obj(
+            streams, ('manifests', lambda _, v: url_or_none(v['url']))
+        ):
+            manifest_type = manifest.get('manifestType')
+            if manifest_type == 'hls':
+                fmts, subs = self._extract_m3u8_formats_and_subtitles(
+                    manifest['url'], video_id, m3u8_id='hls', fatal=False
+                )
+            elif manifest_type == 'dash':
+                fmts, subs = self._extract_mpd_formats_and_subtitles(
+                    manifest['url'], video_id, mpd_id='dash', fatal=False
+                )
+            else:
+                self.report_warning(f'Unsupported manifest type: {manifest_type!r}')
+                continue
+            formats.extend(fmts)
+            self._merge_subtitles(subs, target=subtitles)
+
+        return {
+            'id': video_id,
+            'title': video_info['video']['title'],
+            'description': video_info['video']['descriptionHtml'],
+            'formats': formats,
+            'subtitles': subtitles,
```

Comment:
These should not throw, so wrap them in a `traverse_obj`. Also html should be cleaned (import `clean_html`):
```suggestion
            'formats': formats,
            'subtitles': subtitles,
            **traverse_obj(video_info, ('video', {
                'title': ('title', {str}),
                'description': ('descriptionHtml', {clean_html}),
            },
```